src/pages/history/history.tsx

```diff
@@ -42,8 +41,12 @@ const History = () => {
     } else {
       history.replace(`/history`);
     }
-  }, [history, checkedPrimary, checkedSecondary]);
+  };
+
+  const checkedPrimary = !filter || filter === 'primary';
+  const checkedSecondary = !filter || filter === 'secondary';
 
+  const unconnected = !provider;
```

Comment:
We do use `provider` in many places just to check if we're connected, so the hook could be useful, but in this case:
```ts
const { provider } = useChainData();
const connected = useIsConnected();

if (connected) {
  // TS will still think that provider is possible null here
}
```

Reply:
yeah, but here we do not use the `provider` (at least I saw some places like that while reviewing)... Another solution would be for the hook to return subset of the chain data that is already available when the user is connected. E.g.:
```
const {provider, userAccount} = useIsConnected()
```

but this will not compose with `ourContractXyz` functions... I am not so happy with this idea tbh.